solutions/usaco-922.mdx

```diff
@@ -0,0 +1,95 @@
+---
+id: usaco-922
+source: USACO Gold 2019 February
+title: Dishwashing
+author: Chongtian Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_dishes_gold_feb19.html)
+
+## Alternate Solution - Binary Search
+
+We can binary search for the longest possible prefix. Prefix of length $i-1$ is just prefix of length $i$ without a plate; if is it possible to wash all plates of prefix $i$ then it must also be possible to wash a prefix of $i-1$. Now we only need to simulate the stacking and washing. As the official analysis outlines, if plate $A$ is on top of plate $B$ on the same stack, then $A$ must be less than $B$, or else we cannot access the plate beneath it. Thus, we can greedily place each plate on stack $i$ if the top plate on stack $i$ is larger than our current plate and is the smallest out of all the other soapy stacks. This is because it is always more optimal to satisfy a stricter constraint, rather overriding a broader one if we place the plate on another stack which the plate on top is not the smallest possible.
```

Comment:
can you break this up into paragraphs

Reply:
what 
its not even that much

Reply:
it's really really hard for me to review a giant line of text

Reply:
ok sans just for you <3